### DISCUSSION TITLE
`Enumerable` static method usage in index definition being replaced with LINQ extension method causing an indexing error

### DISCUSSION THREAD

**Discussion by kamranayub** [ravendb/ravendb#13660] [Q&A] [answered]
I am creating a MultiMap index and I have an `AddMap` like this:

```c#
from r in docs
let userId = string.Join("/", Enumerable.Take(r.Id.Split(new[] { '/' }), 2))
```

Setting aside the fact I _could_ pre-compute this expression,when this gets deployed, the index definition ends up changing the expression syntax to `.Take(2)`:

**In the studio**

```c#
.Select(r => new {
    r = r,
    userId = String.Join("/", Id(r).Split(new char[] {
        '/'
    }).Take(2))
})
```

But this throws an indexing error! 😞 

```
Failed to execute mapping function on {docId}. Exception: Microsoft.CSharp.RuntimeBinder.RuntimeBinderException: 'System.Array' does not contain a definition for 'Take'
   at CallSite.Target(Closure , CallSite , Object , Int32 )
```

If I replace the expression _within the Studio_ with `Enumerable.Take(..., 2)` the index works.

What's the best way to handle this? I could not find any guidance in the docs. 😕

**Comment by ppekrol** [comment, marked as answer]
Hi @kamranayub 

We would have to look into that. As a workaround I would suggest doing `.ToList()` after the split. Would that work for you?

**Reply by kamranayub** [reply]
Yep that works!

**Reply by ppekrol** [reply]
Awesome. I've created an issue to look into that: https://issues.hibernatingrhinos.com/issue/RavenDB-18067
